### ISSUE TITLE
preg_replace with /e has been unsupported since PHP 7.0

### ISSUE THREAD

**Issue by elmicha** [returntocorp/sgrep-rules#2506]
**Describe the bug**
The /e modifier of preg_replace has been deprecated since PHP 5.5 and unsupported since PHP 7.0.
PHP 5.5 was released in 2013 and 7.0 in 2015.
Why is this still checked in 2022?

**To Reproduce**
```
<?php
echo preg_replace('/(a)/e', 'strtoupper("\1")', "abc");
```
You can test it on https://3v4l.org/OSZvX#v7.0.0 with the different PHP versions.
Versions since PHP 7.0 do not call the callback function, they produce a warning instead.

**Expected behavior**
Please don't warn about preg_replace, because there is no /e modifier for a long time now.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [X] P2: annoying but not blocking me

**Comment by Sjord** [comment]
This seems to apply to [semgrep-rules/preg-replace-eval.yaml at develop · returntocorp/semgrep-rules · GitHub](https://github.com/returntocorp/semgrep-rules/blob/develop/php/lang/security/preg-replace-eval.yaml).
